### ISSUE TITLE
verify / list seems not to work when rm advertisements present

### ISSUE THREAD

**Issue by willscott** [filecoin-project/index-provider#179]
fails trying to get a block that doesn't exist

**Comment by willscott** [comment]
```
./cmd/provider/provider list --provider-addr-info /ip4/59.138.253.69/tcp/55555/p2p/12D3KooWSPtnggJL5wkoAmtgGCosViEC6Z2heTmt7ZTF98GYPyKq --aerl 1
ID:          baguqeeqqs5l2anifxmmb45hsedz326az24
PreviousID:  baguqeeqqvtphlli7xhee7y5dtzukdohm4a
ProviderID:  12D3KooWSPtnggJL5wkoAmtgGCosViEC6Z2heTmt7ZTF98GYPyKq
Addresses:   [/ip4/59.138.253.69/tcp/55555]
Is Remove:   true
Entries:
  Chunk Count: 0
  Total Count: 0
Note: More entries are available but not synced due to the configured entries recursion limit.
```
```
 ./cmd/provider/provider list --provider-addr-info /ip4/59.138.253.69/tcp/55555/p2p/12D3KooWSPtnggJL5wkoAmtgGCosViEC6Z2heTmt7ZTF98GYPyKq --aerl 2
2022-02-18T15:13:45.927+0100	ERROR	go-legs-dtsync	dtsync/sync.go:170	datatransfer failed	{"cid": "baguqeeqqs5l2anifxmmb45hsedz326az24", "peer": "12D3KooWSPtnggJL5wkoAmtgGCosViEC6Z2heTmt7ZTF98GYPyKq", "message": "data transfer channel QmcqHP5wTfg9QcWDqxYfJ97YdLMv6tTrBi9AAe2c5zF6tD-12D3KooWSPtnggJL5wkoAmtgGCosViEC6Z2heTmt7ZTF98GYPyKq-1645193623922261001 failed to transfer data: channel QmcqHP5wTfg9QcWDqxYfJ97YdLMv6tTrBi9AAe2c5zF6tD-12D3KooWSPtnggJL5wkoAmtgGCosViEC6Z2heTmt7ZTF98GYPyKq-1645193623922261001: graphsync request failed to complete: remote peer is missing block: bafkreehdwdcefgh4dqkjv67uzcmw7oje"}
datatransfer failed
```

**Comment by masih** [comment]
This is caused by removal advertisements that include a list of entries, which are no longer available on the SP side. 

Note that the indexing protocol initially intended to support explicit removal of multihashes in removal ads. But later we decided to only support remove all [here](https://github.com/filecoin-project/index-provider/issues/174).

[This](https://github.com/filecoin-project/index-provider/issues/174) also is a known bug where removal ads are not retrievable due to an internal mapping issue. 

The priority of that ticket was reduced since the assumption was that removal ads are not integrated in lotus yet. But looks like the lotus integration indeed [publishes removal ads](https://github.com/filecoin-project/go-fil-markets/blob/8790cca614d3553b74829bb091afa34b3112f43d/storagemarket/impl/provider_environments.go#L62) when a deal expires or slashed.


Path forward: 
1. Fix #174  
2. In CLI for backward compatibility purposes, ignore entries in removal advertisement and treat them all as "remove all", i.e. do not attempt to list entries in a removal ad. 

Note that removal ad cannot be used as a source of multihashes to perform verify, because its originally advertised are most likely no longer traversable to list the multihashes for verification.
